### ISSUE TITLE
Automatic validation of NATFIXME markers in specs

### ISSUE THREAD

**Issue by herwinw** [natalie-lang/natalie#836]
A lot of specs have a `NATFIXME` comment to describe what needs to be done before they can be enabled again. In many cases, it's a dependency on something external. As an example, in `spec/core/rational/to_r_spec.rb`:
```ruby
it "works when a BasicObject has to_r" do
  obj = BasicObject.new; def obj.to_r; 1 / 2.to_r end
  # NATFIXME: Implement Rational(String)
  # Rational(obj).should == Rational('1/2')
  Rational(obj).should == Rational(1, 2)
end
```
This spec tries to test the call to `to_r`, which works fine, but it tries to validate it using `Rational('1/2')` which has not yet been implemented. As a result, we had to update that statement to `Rational(1, 2)` to perform the same test.
Now when we add String support in `Kernel.Rational`, we would naturally update the spec `spec/shared/rational/Rational.rb`, but since the first one is pretty much unrelated to the actual change we just made, this will probably be forgotten.
Over the past few weeks, I feel like the majority of the Pull Requests I've made are just updates to the specs because we were skipping a lot of specs that were passing by now.

I think it would be better to add automated guards instead of comments, so we would have an option to find those issues easier. Something like this:
```ruby
it "works when a BasicObject has to_r" do
  obj = BasicObject.new; def obj.to_r; 1 / 2.to_r end
  NATFIXME(-> { Rational('1/2') == Rational(1, 2) }) do
    Rational(obj).should == Rational('1/2')
  end
  Rational(obj).should == Rational(1, 2)
end
```
I'm just recycling the name `NATFIXME` to get a uniform way of finding the FIXME markers in the specs. The method takes a callable argument and a block. If the callable argument returns false or raises an error, the block is skipped. Otherwise, the block is valid and we should probably raise an error to let the user know this spec needs an update.
This should work with including other blocks as well:
```ruby
NATFIXME(-> { Rational('1/2') == Rational(1, 2) }) do
  it "works when a BasicObject has to_r" do
    # Actual specs
  end
end
```
We should probably add an envvar to control these blocks, and don't run them by default, so we don't burden people with extra work when they fix something, and one of those NATFIXME blocks would reactivate and then fail again because there is another issue with the block. It's just a way to make it easier to find them.

Does anyone see any problems with this?

**Comment by herwinw** [comment]
Since we're discussing this change in #837 now, I'm closing this issue. It's pretty much a duplicate now.
